lib/language_server/protocol/interface/completion_item.rb

```diff
@@ -1,144 +1,10 @@
-        #
-        # @return [string]
-        def detail
-          attributes.fetch(:detail)
-        end
-
-        #
-        # A human-readable string that represents a doc-comment.
-        #
-        # @return [string]
-        def documentation
-          attributes.fetch(:documentation)
-        end
-
-        #
-        # A string that shoud be used when comparing this item
-        # with other items. When `falsy` the label is used.
-        #
-        # @return [string]
-        def sort_text
-          attributes.fetch(:sortText)
-        end
-
-        #
-        # A string that should be used when filtering a set of
-        # completion items. When `falsy` the label is used.
-        #
-        # @return [string]
-        def filter_text
-          attributes.fetch(:filterText)
-        end
-
-        #
-        # A string that should be inserted a document when selecting
-        # this completion. When `falsy` the label is used.
-        #
-        # @return [string]
-        def insert_text
-          attributes.fetch(:insertText)
-        end
-
-        #
-        # The format of the insert text. The format applies to both the `insertText` property
-        # and the `newText` property of a provided `textEdit`.
-        #
-        # @return [InsertTextFormat]
-        def insert_text_format
-          attributes.fetch(:insertTextFormat)
-        end
-
-        #
-        # An edit which is applied to a document when selecting this completion. When an edit is provided the value of
-        # `insertText` is ignored.
-        #
-        # *Note:* The range of the edit must be a single line range and it must contain the position at which completion
-        # has been requested.
-        #
-        # @return [TextEdit]
-        def text_edit
-          attributes.fetch(:textEdit)
-        end
-
-        #
-        # An optional array of additional text edits that are applied when
-        # selecting this completion. Edits must not overlap with the main edit
-        # nor with themselves.
-        #
-        # @return [TextEdit[]]
-        def additional_text_edits
-          attributes.fetch(:additionalTextEdits)
-        end
-
-        #
-        # An optional command that is executed *after* inserting this completion. *Note* that
-        # additional modifications to the current document should be described with the
-        # additionalTextEdits-property.
-        #
-        # @return [Command]
-        def command
-          attributes.fetch(:command)
-        end
-
-        #
-        # An data entry field that is preserved on a completion item between
-        # a completion and a completion resolve request.
-        #
-        # @return [any]
-        def data
-          attributes.fetch(:data)
-        end
-
-        attr_reader :attributes
-
-        def to_json(*args)
-          attributes.to_json(*args)
-        end
+      class CompletionItem < Base
+        attr_required_keys:label
```

Comment:
ooops

Reply:
Ah, you modified these files manually?

I generated the files by this script:
https://github.com/mtsmfm/language_server-protocol-ruby/blob/fcff46db159de0bba8214ab82abc93b6c940e3e9/scripts/generateFiles.ts

The entrypoint is here:

https://github.com/mtsmfm/language_server-protocol-ruby/blob/fcff46db159de0bba8214ab82abc93b6c940e3e9/bin/generate_files